ui/lib/ldap/addon/components/page/roles.ts

```diff
@@ -8,30 +8,44 @@ import { service } from '@ember/service';
 import { action } from '@ember/object';
 import { getOwner } from '@ember/owner';
 import errorMessage from 'vault/utils/error-message';
+import { tracked } from '@glimmer/tracking';
+import { sanitizePath } from 'core/utils/sanitize-path';
 
 import type LdapRoleModel from 'vault/models/ldap/role';
 import type SecretEngineModel from 'vault/models/secret-engine';
 import type FlashMessageService from 'vault/services/flash-messages';
 import type { Breadcrumb, EngineOwner } from 'vault/vault/app-types';
 import type RouterService from '@ember/routing/router-service';
 import type PaginationService from 'vault/services/pagination';
-import { tracked } from '@glimmer/tracking';
 
 interface Args {
   roles: Array<LdapRoleModel>;
   promptConfig: boolean;
   backendModel: SecretEngineModel;
   breadcrumbs: Array<Breadcrumb>;
   pageFilter: string;
+  roleName?: string;
 }
 
 export default class LdapRolesPageComponent extends Component<Args> {
   @service declare readonly flashMessages: FlashMessageService;
   @service('app-router') declare readonly router: RouterService;
   @service declare readonly pagination: PaginationService;
+
   @tracked credsToRotate: LdapRoleModel | null = null;
   @tracked roleToDelete: LdapRoleModel | null = null;
 
+  // this component renders in the roles.index route and roles.subdirectory (for hierarchal roles)
+  linkParams = (role: LdapRoleModel) => {
+    const route = role.name.endsWith('/') ? 'roles.subdirectory' : 'roles.role.details';
+    // remove trailing forward slash from URL
+    const sanitizedName = sanitizePath(role.name);
```

Comment:
`SanitizePath` does a little more than we're needing it to here. In addition to removing a trailing forward slash it will remove a leading forward slash _and_ whitespace. One idea, might be to create another function in the `addon/utils` directory to remove only the trailing forward slash so we don't have any unexpected behavior. Similar to how we've narrowed down a function to just remove the leading slash called `sanitizeStart`.